tests/functional/api/test_query.py

```diff
@@ -1,9 +1,44 @@
+import pytest
+from pydantic import ValidationError
+
 from ape import chain
+from ape.api.query import AccountQuery, BlockQuery
 
 
 def test_basic_query(eth_tester_provider):
     chain.mine(3)
-    assert chain.blocks.query("number").to_dict() == {"number": {0: 0, 1: 1, 2: 2}}
+    assert chain.blocks.query("number").to_dict() == {"number": {0: 0, 1: 1, 2: 2, 3: 3}}
     df = chain.blocks.query("number", "timestamp")
-    assert len(df) == 3
-    assert df["timestamp"][2] >= df["timestamp"][1] >= df["timestamp"][0]
+    assert len(df) == 4
+    assert df["timestamp"][3] > df["timestamp"][2] >= df["timestamp"][1] >= df["timestamp"][0]
+    df_all = chain.blocks.query("*")
+    columns = list(df_all.columns)
+    assert columns == [
+        "gas_data",
+        "consensus_data",
+        "hash",
+        "number",
+        "parent_hash",
+        "size",
+        "timestamp",
+    ]
+
+
+def test_block_query(eth_tester_provider):
```

Comment:
In general, it is better to use multiple test cases.

Tests should test one thing. Now, if one part of the tests fails, the whole test fails, and that makes it harder to pinpoint what is actually failing.